src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryCachingConfiguration.java

```diff
@@ -97,8 +97,17 @@ public FormValidation doClearCache(@QueryParameter String name) throws Interrupt
                         if (libraryNamePath.readToString().equals(name)) {
                             FilePath libraryCachePath = LibraryCachingConfiguration.getGlobalLibrariesCacheDir()
                                     .child(libraryNamePath.getName().replace("-name.txt", ""));
-                            libraryCachePath.deleteRecursive();
-                            libraryNamePath.delete();
+                            ReentrantReadWriteLock retrieveLock = LibraryAdder.getReadWriteLockFor(libraryCachePath.getName());
+                            if (retrieveLock.writeLock().tryLock(10, TimeUnit.SECONDS)) {
+                                try {
+                                    libraryCachePath.deleteRecursive();
+                                    libraryNamePath.delete();
+                                } finally {
+                                    retrieveLock.writeLock().unlock();
+                                }
+                            } else {
+                                return FormValidation.error("The cache dir is currently used by another thread, so deletion was not possibly. Please try again");
```

Comment:
Rather than "possibly" in the error it should read "possible". It may also be more clear to the user to start with the failure and then follow with the explanation. For example, "The cache dir could not be deleted because it is currently being used by another thread. Please try again"